packages/amesos2/src/Amesos2_Superlu_FunctionMap.hpp

```diff
@@ -724,11 +737,8 @@ namespace Amesos2 {
 
 #ifdef HAVE_TEUCHOS_COMPLEX
 
-  /* The specializations for Teuchos::as<> for SLU::complex and
-   * SLU::doublecomplex are provided in Amesos2_Superlu_Type.hpp
-   */
   template <>
-  struct FunctionMap<Superlu,SLU::C::complex>
+  struct FunctionMap<Superlu, Kokkos::complex<float>>
```

Comment:
Are you saying SuperLU can take Kokkos::complex directly, there was some problem with it long time ago, that is why we we needed its inbuilt complex. I don't remember what it is. If it works w/ kokkos::complex then that is fine to skip the old one.

Reply:
This seemed fine but I did't know about the prior issues. So maybe I am not seeing a problem that is going to come up for some build.

Reply:
Switched back to manual copy to an array of native SLU type, so no more use of reinterpret cast.